packages/fetchai/connections/webhook/connection.py

```diff
@@ -0,0 +1,267 @@
+        self.loop = (
+            None
+        )  # type: Optional[asyncio.AbstractEventLoop]  # pragma: no cover
+        logger.info("Initialised a webhook channel")
+
+    async def connect(self):
+        """
+        Connect the webhook
+
+        Connects the webhook via the webhook_address and webhook_port parameters
+        """
+        if self.is_stopped:
+            self.app = web.Application()
+            self.app.add_routes(
+                [web.post(self.webhook_url_path, self._receive_webhook)]
+            )
+            self.runner = web.AppRunner(self.app)
+            await self.runner.setup()
+            self.webhook_site = web.TCPSite(
+                self.runner, self.webhook_address, self.webhook_port
+            )
+            await self.webhook_site.start()
+            self.is_stopped = False
+
+    async def disconnect(self):
+        """
+        Disconnect.
+
+        Shut-off and cleanup the webhook site, the runner and the web app, then stop the channel.
+        """
+        assert (
+            self.webhook_site is not None
+        ), "Application not connected, call connect first!"
+
+        if not self.is_stopped:
+            logger.info("Webhook app is shutdown.")
+            await self.webhook_site.stop()
+            await self.runner.shutdown()
+            await self.runner.cleanup()
+            await self.app.shutdown()
+            await self.app.cleanup()
+            self.is_stopped = True
+
+    async def _receive_webhook(self, request: web.Request):
+        """
+        Receive a webhook request
+
+        Get webhook request, turn it to envelop and send it to the agent to be picked up.
+        """
+        webhook_envelop = await self.to_envelope(self.connection_id, request)
+        self.in_queue.put_nowait(webhook_envelop)  # type: ignore
+        return web.Response(status=200)
+
+    def send(self, request_envelope: Envelope) -> None:
+        pass
+
+    async def to_envelope(
+        self, connection_id: PublicId, request: web.Request,
+    ) -> Envelope:
+        """
+        Convert a webhook request object into an Envelope containing an HttpMessage (from the 'http' Protocol).
+
+        :param connection_id: the connection id
+        :param request: the webhook request
+        """
+
+        payload_bytes = await request.read()
+        version = str(request.version[0]) + "." + str(request.version[1])
+
+        context = EnvelopeContext(connection_id=connection_id)
+        http_message = HttpMessage(
+            performative=HttpMessage.Performative.REQUEST,
+            method=request.method,
+            url=str(request.url),
+            version=version,
+            headers=json.dumps(dict(request.headers)),
+            bodyy=payload_bytes if payload_bytes is not None else b"",
+        )
+        envelope = Envelope(
+            to=self.agent_address,
+            sender=request.remote,
```

Comment:
What does `request.remote` return?

Reply:
From docstring of the method:

"Remote IP of client initiated HTTP request"